src/hooks/useLoginProfile.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  CodeChallengeMethod,
+  exchangeCodeAsync,
+  makeRedirectUri,
+  refreshAsync,
+  ResponseType,
+  useAuthRequest
+} from 'expo-auth-session';
+import { useEffect, useState } from 'react';
+
+import * as appJson from '../../app.json';
+import { secrets } from '../config';
+import { addToStore, readFromStore } from '../helpers';
+
+const PROFILE_ACCESS_TOKEN = 'profileAccessToken';
+
+const namespace = appJson.expo.slug as keyof typeof secrets;
+const profile = secrets[namespace]?.profile;
+const serverUrl = profile?.serverUrl;
+
+const keycloakDiscovery = {
+  authorizationEndpoint: serverUrl + '/auth',
+  tokenEndpoint: serverUrl + '/token',
+  revocationEndpoint: serverUrl + '/revoke'
+};
+
+const CLIENT_ID = profile?.clientId;
+const CLIENT_SECRET = profile?.clientSecret;
+const SCOPES = ['openid', 'email', 'profile', 'phone', 'address'];
+
+export const useLoginProfile = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const redirectUri = makeRedirectUri({
+    scheme: appJson.expo.scheme,
+    path: 'redirect'
+  });
+
+  const [request, response, promptAsync] = useAuthRequest(
+    {
+      clientId: CLIENT_ID,
+      codeChallengeMethod: CodeChallengeMethod.S256,
+      redirectUri,
+      responseType: ResponseType.Code,
+      scopes: SCOPES,
+      usePKCE: false
+    },
+    keycloakDiscovery
+  );
+
+  useEffect(() => {
+    const handleAuthResponse = async () => {
+      if (response?.type === 'success') {
+        const { code } = response.params;
+        await exchangeToken(code);
+      }
+    };
+
+    handleAuthResponse();
+  }, [response]);
+
+  const exchangeToken = async (code: string) => {
+    try {
+      setLoading(true);
+      const tokenResult = await exchangeCodeAsync(
+        {
+          code,
+          clientId: CLIENT_ID,
+          clientSecret: CLIENT_SECRET,
+          redirectUri,
+          extraParams: {
+            grant_type: 'authorization_code'
+          }
+        },
+        keycloakDiscovery
+      );
+
+      setIsLoggedIn(true);
+      await addToStore(PROFILE_ACCESS_TOKEN, JSON.stringify(tokenResult));
+    } catch (error) {
+      console.error('Error exchanging code:', error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    (async () => {
+      await checkToken();
+    })();
+  }, []);
```

Comment:
wouldn't `useEffect(checkToken, []);` work here? why would we need to await something if we are not doing anything with its returned value?

Reply:
This only enables the checkToken function to work. checkToken checks whether there is a token stored in the device and whether this token is expired. If the token has expired, a new token is automatically generated and stored in the device. 

But yes, you are right, if we don't need the value returned from the function, then the async/await structure can be removed.

Reply:
In my opinion, I see no reason why this code should not work. But if I update the code as you say, I get this error. Therefore, I think it should remain as it is now.

<img width="461" alt="image" src="https://github.com/user-attachments/assets/b8cb76f2-2fdb-493a-b9ba-ea55fe50c424" />

Reply:
huh, those are the same things. weird. and nothing has to do with some `create`, right? can you share the git diff of that change?

Reply:
I only have this modification. I have no idea why the error message mentions `create`.

<img width="545" alt="image" src="https://github.com/user-attachments/assets/0075513b-d84d-4404-baa0-722c1887d201" />


The issue was caused by passing an async function directly to useEffect like this: 
```
  useEffect(checkToken, []);
```
Since React doesn’t allow async functions as effect callbacks (because they return a Promise), it led to a runtime error.That's why we have to leave the code as it is.